main/channel.c

```diff
@@ -1429,20 +1479,41 @@ struct ast_channel *ast_channel_get_by_name_prefix(const char *name, size_t name
 {
 	struct ast_channel *chan;
 	char *l_name = (char *) name;
+	struct ast_channel_callback_data callback_data = {.search = name, .final_match_result = 0};
 
 	if (ast_strlen_zero(l_name)) {
 		/* We didn't have a name to search for so quit. */
 		return NULL;
 	}
 
-	chan = ast_channel_callback(ast_channel_by_name_cb, l_name, &name_len,
+  try_name_again:
```

Comment:
Same thing here, you can use 2 loops here and goto isn't needed.

Reply:
Are goto's that bad?  It reduces nesting.  The only thing that adding a loop does is introduce an extra level of nesting.

Reply:
They're not inherently bad, this just doesn't seem like a necessary or appropriate use for them, since a loop works perfectly fine. I don't think there's a problem with adding a level of nesting, it contains the logic in its own scope.

Reply:
Fixed